addons/l10n_in_edi/models/account_move.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+import json
+
+from odoo import fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_in_edi_cancel_reason = fields.Selection(
+        selection=[
+            ("1", "Duplicate"),
+            ("2", "Data Entry Mistake"),
+            ("3", "Order Cancelled"),
+            ("4", "Others"),
+        ],
+        string="Cancel reason",
+        copy=False,
+    )
+    l10n_in_edi_cancel_remarks = fields.Char("Cancel remarks", copy=False)
+
+    def _extract_digits(self, string):
+        if not string:
+            return string
+        matches = re.findall(r"\d+", string)
+        result = "".join(matches)
+        return result
+
+    def button_cancel_posted_moves(self):
+        """Mark the edi.document related to this move to be canceled."""
+        reason_and_remarks_not_set = self.env["account.move"]
+        is_cancelled = False
+        for move in self:
+            send_l10n_in_edi = move.edi_document_ids.filtered(
+                lambda doc: doc.edi_format_id.code == "in_einvoice_1_03" and doc.state == "sent"
+            )
+            if send_l10n_in_edi:
+                is_cancelled = True
+            # check submitted E-invoice does not have reason and remarks
+            # because it's needed to cancel E-invoice
+            if send_l10n_in_edi and (not move.l10n_in_edi_cancel_reason or not move.l10n_in_edi_cancel_remarks):
+                reason_and_remarks_not_set += move
+        if reason_and_remarks_not_set:
+            raise UserError(
+                _(
+                    "To cancel E-invoice set cancel reason and remarks in invoices: \n%s",
+                    ("\n".join(reason_and_remarks_not_set.mapped("name"))),
+                )
+            )
+        res = super().button_cancel_posted_moves()
+        if is_cancelled:
+            # it's must cancel within 24 hours else user can't cancel the invoice
+            # so trigger web service immediately.
+            self.env.ref("account_edi.ir_cron_edi_network")._trigger()
```

Comment:
It should be triggered automatically.  No need.

Reply:
okay, removing